curp/src/server.rs

```diff
@@ -1,39 +1,129 @@
-use std::{ops::Not, sync::Arc};
+use std::{collections::VecDeque, fmt::Debug, net::SocketAddr, ops::Not, sync::Arc};
 
-use parking_lot::Mutex;
+use madsim::net::Endpoint;
+use parking_lot::{Mutex, RwLock};
 
 use crate::{
-    cmd::{Command, CommandExecutor},
+    cmd::{CmdId, Command, CommandExecutor},
+    error::ServerError,
+    log::LogEntry,
     message::{Propose, ProposeError, ProposeResponse},
     util::MutexMap,
 };
 
+/// Default server serving port
+pub(crate) static DEFAULT_SERVER_PORT: &str = "12345";
+
+/// The Rpc Server to handle rpc requests
+#[derive(Clone, Debug)]
+pub struct RpcServerWrap<C: Command + 'static, CE: CommandExecutor<C> + 'static> {
+    /// The inner server wrappped in an Arc so that we share state while clone the rpc wrapper
+    inner: Arc<Server<C, CE>>,
+}
+
+#[allow(missing_docs)]
+#[madsim::service]
+impl<C: Command + 'static, CE: CommandExecutor<C> + 'static> RpcServerWrap<C, CE> {
+    /// handle Propose
+    #[rpc]
+    async fn propose(&self, p: Propose<C>) -> ProposeResponse<C> {
+        self.inner.propose(p).await
+    }
+
+    /// Run a new rpc server
+    #[allow(dead_code)]
+    async fn run(is_leader: bool, term: u64, others: &[SocketAddr]) -> Result<(), ServerError> {
+        let rx_ep = Endpoint::bind(format!("0.0.0.0:{DEFAULT_SERVER_PORT}")).await?;
```

Comment:
servers use same port then we cannot test on same machine.

Reply:
Let me add a customized port option.